tensorboard/webapp/alert/types.ts

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+/**
+ * An alert structure used when creating newly reported alerts.
+ */
+export interface AlertReport {
+  localizedMessage: string;
```

Comment:
I may have left a comment in the other PR but I don't think I find any comment.

How would you plan on supporting different actions in the snackbar?  i.e., in some cases, we want to have button to retry sending a request. In order to replace our current error handling, we would need that facility. Without it, we will have two ways to achieve the same thing.

Reply:
I'd like to do this in a separate PR, if that works for you.  This is what I had in mind:

extend `AlertReport` with a new field:
```
export interface AlertReport {
  localizedMessage: string;
  retryCallback?: (store: Store) => void;
}
```

used like so:
```
AlertActionModule.registerActionAlerts([
  {
    localizedMessage: "Fetch failed",
    retryCallback: (store) => {
      // ...
      store.dispatch(fetchAction(...));
    }
  },
])
```

with the AlertSnackbarContainer using `.openFromComponent`, instead of `.open`,
to open an AlertSnackbarComponent, which contains:

```
// template
<div *ngIf="!!alertInfo.retryCallback" (click)="onRetryClicked">Retry</div>

// component
@Input() alertInfo: AlertInfo

onRetryClicked() { this.alertInfo.retryCallback(); }
```

Reply:
I wish it to be more generic than just retryCallback (and it, I don't think should be a callback since it opens the API for just about anything) but I will look forward to seeing the change afterwards.